### PR TITLE
[PB-2979]: fix/share dialog and use the new recents endpoint

### PR DESCRIPTION
**Pull request by xabg2**
- Fix the Shared dialog for items - Show the password if the item has
- Use the new Recents endpoint
- Display the shared icon for shared items in Recents

### CODE REVIEW

**Comment by xabg2** [comment]
⚠️  Depends on the following PRs:
- SDK: https://github.com/internxt/sdk/pull/249
- Drive server WIP: https://github.com/internxt/drive-server-wip/pull/409

**Update:**
- This PR is ready to merge

File: src/app/i18n/locales/en.json [hunk 1]
```diff
@@ -1303,7 +1303,6 @@
     "invalidTaxIdIsProvided": "The provided Tax ID is invalid",
     "errorWhileFetchingMoreItems": "Something went wrong while fetching more items",
     "errorDeletingItems": "Something went wrong while deleting this item",
-    "errorApplyingCoupon": "Something went wrong while applying the coupon",
```
Comment (CandelR) [inline src/app/i18n/locales/en.json:1306]: why has this translation been removed?

File: src/app/drive/components/DriveExplorer/DriveExplorer.tsx [hunk 2]
```diff
@@ -527,10 +529,14 @@ const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
...
+    dispatch(fetchSortedFolderContentThunk(currentFolderId));
+  };
+
   const handleOnShareItem = useCallback(() => {
     setTimeout(() => {
-      resetPaginationState();
-      dispatch(fetchSortedFolderContentThunk(currentFolderId));
+      fetchFolderContent?.() ?? resetPaginationStateAndFetchDriveFolderContent(currentFolderId);
     }, 500);
   }, [currentFolderId]);
```
Comment (CandelR) [inline src/app/drive/components/DriveExplorer/DriveExplorer.tsx]: Remember to add `fetchFolderContent` to dependency array

**Review by CandelR — CHANGES_REQUESTED** [review]

File: src/app/i18n/locales/en.json (hunk 1 above)
Comment (xabg2) [reply src/app/i18n/locales/en.json:1306]: It was duplicated. You can find it above the `invalidTaxIdsProvided` error.

**Review by CandelR — APPROVED** [review]

**Review by Jona-Internxt — APPROVED** [review]

**Pull request merged**